listen/property-changes.js

```diff
@@ -57,41 +55,95 @@ module.exports = PropertyChanges;
 function PropertyChanges() {
     throw new Error("This is an abstract interface. Mix it. Don't construct it");
 }
+var PropertyChangesP = PropertyChanges.prototype;
+
+function ObjectPropertyChangeDescriptor(propertyName) {
+	this.propertyName = propertyName;
+	return this;
+}
+Object.defineProperties(ObjectPropertyChangeDescriptor.prototype,{
+	propertyName: {
+		value:null,
+		writable: true
+	},
+	_willChangeListeners: {
+		value:null,
+		writable: true
+	},
+	willChangeListeners: {
+		get: function() {
+			return this._willChangeListeners || (this._willChangeListeners = new WillChangeListenersRecord(this.propertyName));
+		}
+	},
+	_changeListeners: {
+		value:null,
+		writable: true
+	},
+    changeListeners: {
+		get: function() {
+			return this._changeListeners || (this._changeListeners = new ChangeListenersRecord(this.propertyName));
+		}
+	}
+});
+
+function ChangeListenersRecord(propertyName) {
+    this.specificHandlerMethodName = "handle" + propertyName + "Change";
+	return this;
+}
+
+Object.defineProperties(ChangeListenersRecord.prototype,{
+    _current: {
+		value: null,
+		writable: true
+	},
+	current: {
+		get: function() {
+			return this._current || (this._current = []);
+		},
+	},
+    _active: {
+		value: null,
+		writable: true
+	},
+    active: {
+		get: function() {
+			return this._active || (this._active = []);
+		}
+	},
+    genericHandlerMethodName: {
+		value: "handlePropertyChange"
+	}
+});
+
+function WillChangeListenersRecord(propertyName) {
+    this.specificHandlerMethodName = "handle" + propertyName + "WillChange";
+	return this;
+}
+WillChangeListenersRecord.prototype = new ChangeListenersRecord();
+WillChangeListenersRecord.prototype.genericHandlerMethodName = "handlePropertyWillChange";
+
+
 
 PropertyChanges.debug = true;
 
-PropertyChanges.prototype.getOwnPropertyChangeDescriptor = function (key) {
-    if (!this.__propertyChangeListeners__) {
+PropertyChangesP.__propertyChangeListeners__ = null;
+
+PropertyChangesP.getOwnPropertyChangeDescriptor = function (key) {
```

Comment:
Please avoid using this *P convention for prototypes. Please do not abbreviate or create acronyms for the clarity of the code.